### DISCUSSION TITLE
Getting started for existing apps

### DISCUSSION THREAD

**Discussion by ladeak** [microsoft/aspire#5554] [Q&A] [answered]
Given I have an app with with a dependency (say Kafka messaging). I would like to add an aspire host that starts Kafka in a container my app configured to use Kafka. However, I have my own appsettings json for the configuration, which I would like to keep using. How can I map the Kafka settings onto my projects appsettings ?

**Comment by davidfowl** [comment, marked as answer]
Using WithEnvironment instead of WithReference you can change the format of the name so it matches your configuration. Assuming you are using .NET's IConfiguration system, that should just work.

**Reply by ladeak** [reply]
Thank you
